x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/dataframe/DataFrameAnalyticsManager.java

```diff
@@ -294,6 +294,10 @@ private void reindexDataframeAndStartAnalysis(DataFrameAnalyticsTask task, DataF
                 new GetIndexRequest().indices(config.getDest().getIndex()), destIndexListener);
     }
 
+    private boolean isTaskCancelledException(Exception error) {
+        return ExceptionsHelper.unwrapCause(error) instanceof TaskCancelledException || error.getCause() instanceof TaskCancelledException;
```

Comment:
I wonder if it's also worth unwrapping the cause?

```suggestion
        return ExceptionsHelper.unwrapCause(error) instanceof TaskCancelledException || ExceptionsHelper.unwrapCause(error.getCause()) instanceof TaskCancelledException;
```

Whether this is worth it or not depending on whether the nested exception could be a remote transport exception, or whether only the outermost exception can be.  If you don't know the answer it's probably better to add the extra unwrap call.

Reply:
Good point. I'll do that.